setup.py

```diff
@@ -869,6 +875,148 @@ def prefix(directory, files):
     extra_objects=zstd_extra_objects,
 )
 
+# SZ3 library and its hdf5 filter
+# Generate needed version include file.
+with open(os.path.join("src", "SZ3", "include", "SZ3", "version.hpp"), 'w') as f:
+    f.write("//\n")
+    f.write("// DO NOT MODIFY version.hpp, it is automatically generated by cmake\n")
+    f.write("// The version is controlled in CMakeLists.txt\n")
+    f.write("//\n")
+    f.write("\n")
+    f.write("#ifndef SZ3_VERSION_HPP\n")
+    f.write("#define SZ3_VERSION_HPP\n")
+    f.write("\n")
+    f.write('#define SZ3_NAME "SZ3"\n')
+    f.write('#define SZ3_VER  "3.1.7"\n')
+    f.write("#define SZ3_VER_MAJOR 3\n")
+    f.write("#define SZ3_VER_MINOR 1\n")
+    f.write("#define SZ3_VER_PATCH 7\n")
+    f.write("#define SZ3_VER_TWEAK \n")
+    f.write("\n")
+    f.write("#endif //SZ3_VERSION_HP\n")
+
+sz3_dir = os.path.join("src", "SZ3")
+sz3_sources = [] #glob(os.path.join(sz3_dir, "tools", "sz3", "sz3.*pp"))
+sz3_include_dirs = [os.path.join(sz3_dir, "include"),
+                    os.path.join(sz3_dir, "include", "SZ3"),
+                    os.path.join(sz3_dir, "include", "SZ3", "api"),
+                    os.path.join(sz3_dir, "include", "SZ3", "compressor"),
+                    os.path.join(sz3_dir, "include", "SZ3", "encoder"),
+                    os.path.join(sz3_dir, "include", "SZ3", "frontend"),
+                    os.path.join(sz3_dir, "include", "SZ3", "lossless"),
+                    os.path.join(sz3_dir, "include", "SZ3", "predictor"),
+                    os.path.join(sz3_dir, "include", "SZ3", "preprocessor"),
+                    os.path.join(sz3_dir, "include", "SZ3", "quantizer"),
+                    os.path.join(sz3_dir, "include", "SZ3", "utils"),
+                    os.path.join(sz3_dir, "include", "SZ3", "utils", "inih"),
+                    os.path.join(sz3_dir, "include", "SZ3", "utils", "ska_hash"),
+                    ]
+if sys.platform.startswith('darwin'):
+    # provide dummy omp.h
+    sz3_include_dirs.append(os.path.join("src", "SZ3_extra", "darwin"))
+
+sz3_hdf5_dir = os.path.join(sz3_dir, "tools", "H5Z-SZ3")
+
+# I have issues with the BMI on linux and Mac because of compiling Zstd separately
+
+if sys.platform.startswith('darwin'):
+    HDF5PLUGIN_ZSTD_FROM_BLOSC = False
+else:
+    HDF5PLUGIN_ZSTD_FROM_BLOSC = True
+
+if not HDF5PLUGIN_ZSTD_FROM_BLOSC:
+    sz3_zstd_cmake_list = [
```

Comment:
May simpler to use `glob` here as for zstd for blosc.

Reply:
I cut and pasted from their CMakeLists file for the shipped Zstd code.